### ISSUE TITLE
Wrong error message when an attribute is a substring of another

### ISSUE THREAD

**Issue by boris-petrov** [typed-ember/glint#186]
Component `Foo`:

```typescript
import Component from '@glint/environment-ember-loose/glimmer-component';

export interface IFoo {
  readonly Args: {
    readonly a: number;
    readonly aa: string;
  };
}

export default class Foo extends Component<IFoo> {}

declare module '@glint/environment-ember-loose/registry' {
  export default interface Registry {
    Foo: typeof Foo;
  }
}
```

Component `Bar`:

```typescript
import Component from '@glint/environment-ember-loose/glimmer-component';

export default class Bar extends Component {}
```

```handlebars
<Foo @aa="asd" @a="qwe" />
```

Leads to:

```
app/components/bar.hbs:1:7 - error TS2322: Type 'string' is not assignable to type 'number'.

1 <Foo @aa="asd" @a="qwe" />
        ~

  app/components/foo.ts:5:14
    5     readonly a: number;
                   ~
    The expected type comes from property 'a' which is declared here on type '{ readonly a: number; readonly aa: string; }'
```

Note the first squiggly line - it's on `aa` instead of on `a`.

**Comment by boris-petrov** [comment]
@dfreeman - there are two more cases where this breaks:

1)

```hbs
{{foo aa="qwe" a="qwe"}}
```

```
app/bar.hbs:1:7 - error TS2322: Type 'string' is not assignable to type 'number'.

1 {{foo aa="qwe" a="qwe"}}
        ~

  app/foo.ts:5:14
    5     readonly a: number;
                   ~
    The expected type comes from property 'a' which is declared here on type '{ readonly a: number; readonly aa: string; }'
```

2)

```hbs
<Foo
  @aa={{@a}}
  @a="qwe" />
```

Where `bar` now has this definition:

```ts
export default class Bar extends Component<{ Args: { a: string } }> {}
```

```
app/bar.hbs:2:10 - error TS2322: Type 'string' is not assignable to type 'number'.

2   @aa={{@a}}
           ~

  app/foo.ts:5:14
    5     readonly a: number;
                   ~
    The expected type comes from property 'a' which is declared here on type '{ readonly a: number; readonly aa: string; }'
```

**Comment by dfreeman** [comment]
It would be so nice if `@glimmer/syntax` would give us the actual offsets for attrs rather than us having to reverse-engineer them 🙃 

Thanks for the additional report!

**Comment by dfreeman** [comment]
FYI for anyone looking at this: it's still on our radar! We'd like to shift to using `@glimmer/syntax`'s `ASTv2` parse format, which still doesn't give location info for 100% of identifiers, but definitely covers more than v1 and should make these situations much rarer.
